yt/frontends/boxlib/data_structures.py

```diff
@@ -1405,6 +1406,14 @@ def _read_header(raw_file, field):
                 lo_corner, hi_corner, node_type = _line_to_numpy_arrays(clean_line)
                 boxes.append((lo_corner, hi_corner, node_type))
 
+            if nghost is None:
```

Comment:
I would suggest to only parse `nghost_lines`* here once `lo_corner` is defined, in order to make the process a bit easier to follow. To be clear, my suggestion is to move the previous try block here and merge it with this `if` statement.
Furthermore, I think it'd be clearer to use a variable here , e.g. `ndim = len(lo_corner)`

* come to think of it the name of this variable is also misleading since it really refers to a _single_ line being parsed into _several_ values... can you also rename it while you're at it ?

Reply:
Thanks for the comments - changes made. Did you mean to rename `nghost_lines`?

Reply:
exactly, and you nailed it !